src/application_input.hpp

```diff
@@ -0,0 +1,54 @@
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef APPLICATION_INPUT_HPP_
+#define APPLICATION_INPUT_HPP_
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "defs.hpp"
+#include "interface/properties_interface.hpp"
+#include "interface/state_descriptor.hpp"
+#include "parameter_input.hpp"
+#include "parthenon_arrays.hpp"
+
+namespace parthenon {
+
+struct ApplicationInput {
+ public:
+  // ParthenonManager functions
+  std::function<void()> SetFillDerivedFunctions = nullptr;
```

Comment:
Ah, it's a little too late for this, I guess, but I think this is a good candidate for just being a virtual base class rather than using `std::function`s, unless there's some optimization that can be taken if you know the user doesn't provide a given function.

If you _were_ to change it, I would provide default implementations of all these that do nothing, and then users just override the routines that they need to.